### PR TITLE
Supplyborgs can now close closets and crates

### LINKED ISSUE #25325: Supply Cyborgs can't close containers
#### Description of issue

Supply Cyborgs are incapable of basic Cargo Technician role functionality - closing supply crates. This renders Supply Cyborgs not capable of working in this branch of Supply role.

image

#### Difference between expected and actual behavior

Supply Cyborgs should be capable of closing crates and/or containers in general.

#### Steps to reproduce

* Spawn as a Supply Cyborg
* Find an open crate or pry it open with a crowbar.
* There's no way to close it with any obvious or non-obvious means
  * Trying to close it with Toggle Open verb returns "this mob can't use that verb type"

#### Length of time in which bug has been known to occur

Unknown

#### Server revision

* **Game:** c052e422c7311c2a8cbddf0dec1fd6afcddac7a2 
* **Patches:** fc6cfdf0edbca13310b472292d5628b92dced134 

[bugfix][oversight][easy]

### PR DESCRIPTION
**Pull request by Faptastic**
Fixes #25325
Adding a clipboard was also requested in the thread, didn't see a problem with it.
The toggle_open verb will also work for them, its suprising just how much stuff can't use that but should be able to.
Theres a comment claiming attack_robot is broken but using attack_ai I managed to break everything so... yeah. I'm hoping thats not going to be an issue because I tried it both ways.
Tested using human, mommi, robot (not supply),robot (supply) and even ai and it seems to be working fine.

ezgif com-optimize
[featurerequest][qol]
:cl:
 * rscadd: Supply borgs get a clipboard. They can also open and close crates/lockers.

### CODE REVIEW

**Comment by RaccoonBro** [comment]
Are they able to lock lockers?

**Comment by Faptastic** [comment]
I'm pretty sure borgs have always been able to do that using modules that don't have an effect on the locker like wrapping paper would as long as its closed, especially the crowbar.
The toggle lock verb doesn't work for borgs but then it doesn't work for a few mobs that have the right to do so. Supplyborgs still can't actually crowbar them open either way but can open with a click or a verb, crowbar code manages that I assume.
Humans can relock by slapping them with objects as well providing they're wearing their id and have the right access, or slapping with said id.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -576,13 +576,17 @@
 	if(usr.incapacitated())
 		return
 
-	if(ishuman(usr) || isMoMMI(usr))
-		if(isMoMMI(usr))
-			src.add_hiddenprint(usr)
-			add_fingerprint(usr)
+	if (isrobot(usr))
```
Comment (ShiftyRail) [inline code/game/objects/structures/crates_lockers/closets.dm:579]: This isn't your fault (I assume you copypasted this bit from somewhere else), but do not use `usr`. Use `user` instead.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 2]
```diff
@@ -576,13 +576,17 @@
...
+			src.attack_robot(R)
+			return
+
+	if(ishuman(usr))
 		src.attack_hand(usr)
-	else
-		to_chat(usr, "<span class='warning'>This mob type can't use this verb.</span>")
+		return
+
+	to_chat(usr, "<span class='warning'>This mob type can't use this verb.</span>")
```
Comment (ShiftyRail) [inline code/game/objects/structures/crates_lockers/closets.dm]: Maybe a more IC-friendly message while we're at it ?

File: code/game/objects/structures/crates_lockers/closets.dm (hunk 1 above)
Comment (Faptastic) [reply code/game/objects/structures/crates_lockers/closets.dm:579]: I was taught about that but it's being used in a verb not a proc so user is undefined as is? Unless you can't pass it as a parameter to a proc either in which case I would need to define it as a mob before checking its type and go from there?

Comment (Kurfursten) [reply code/game/objects/structures/crates_lockers/closets.dm:579]: it's okay in a verb, I think shifty was confused by the way the diff presents this

**Pull request merged**